### PR TITLE
Allow to disable plugins by default

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: utils/common/src/main/kotlin/PluginManager.kt [hunk 1]
```diff
@@ -46,6 +46,12 @@ interface Plugin {
      * The type of the plugin.
      */
     val type: String
+
+    /**
+     * A flag to indicate whether the plugin should be enabled by default or not.
+     */
+    val isEnabledByDefault: Boolean
```
Comment (fviernau) [inline utils/common/src/main/kotlin/PluginManager.kt:53]: This property needs a mechanism for enabling a plugin. IIUC such mechanism is implemented only for  package mangager plugins. Would it thus make sense to move this property to `PackageManagerFactory`?

**Review by fviernau — CHANGES_REQUESTED** [review]

File: utils/common/src/main/kotlin/PluginManager.kt (hunk 1 above)
Comment (sschuberth) [reply utils/common/src/main/kotlin/PluginManager.kt:53]: > This property needs a mechanism for enabling a plugin.

Why? They're enabled by default.

> Would it thus make sense to move this property to `PackageManagerFactory`?

No, this should be a generic aspect of any plugin. In the future, I see more plugin types making use of this feature.

File: utils/common/src/main/kotlin/PluginManager.kt [hunk 2]
```diff
@@ -46,6 +46,12 @@ interface Plugin {
      * The type of the plugin.
      */
     val type: String
+
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply utils/common/src/main/kotlin/PluginManager.kt:49]: I'm soon about to create a PR for the new `GradleInspector` implementation. While I've tested it thoroughly, I don't want it to replace the "classic" `Gradle` analyzer right from the start. However, not both Gradle analyzers should be enabled at the same time, as they would discover the same definition files, and create an analyzer result of double the size. For comparing the implementations it's anyway much more convenient to do separate runs per implementation, and *then* compare the individual result files. That's why I want the new `GradleInspector` implementation to be available in the code base, but disabled by default. Users can then conveniently enable it via the `ort analyze` command line option.

[the remark this answers is not in the document]
Comment (sschuberth) [reply utils/common/src/main/kotlin/PluginManager.kt:49]: To clarify even more, a package manager plugin implementation that's supposed to be disabled by default would then simply do

    class Factory : AbstractPackageManagerFactory<GradleInspector>("GradleInspector", isEnabledByDefault = false)

[the remark this answers is not in the document]
Comment (fviernau) [reply utils/common/src/main/kotlin/PluginManager.kt:49]: Thanks!

**Pull request merged**